### PR TITLE
Rename "Boa" to boa_engine, moved GC and profiler to their crates

### LINKED ISSUE #230: Rename cargo package
Rename `Boa` to `boa`

### PR DESCRIPTION
**Pull request by Razican**
This PR closes #230.

It changes the following:

- Changes the main crate name from "Boa" to "boa_engine"
- Creates the "boa_gc" crate which for now just exports the "gc" crate, but will contain a custom garbage collector in the future. Unfortunately, due to how the "gc_derive" crate is implemented, we still need to import the "gc" crate in "boa_engine"
- Creates the "boa_profiler" crate with the boa profiler.
- Improves some imports, cleans some stuff up (for example, we were re-exporting the `Position` in multiple places, or we were importing the `TryInto` trait, which is now in the prelude)
- Ordered workspace member crates alphabetically
- Added more exclusions to package creations
- Improved categories, keywords and ordered Cargo.toml fields
- Added new folders to dependabot (and boa_unicode, which was missing)
- Fixed some caching inconsistencies
- Removed top-level re-exports for external crates. Let me know if you want me to add this back :)

I didn't move the Boa console to its own crate, since this requires further work in the engine, since it is currently part of the `Context`. Ideally, I would like this to be a separate crate and for "boa_engine" to not have it as a dependency ("boa_cli" would depend on it, of course, maybe "boa_wasm" too).

### CODE REVIEW

**Comment by Razican** [comment]
Unfortunately, the benchmarking error won't go away, since the path for the main crate is different in the PR and in the "main" branch.

**Review by RageKnify — APPROVED** [review]

**Review by jedel1043 — APPROVED** [review]
No comments from my side :)

**Review by raskad — CHANGES_REQUESTED** [review]
Looks good to me.
Just the fixes for `boa_profiler`.

**Review by HalidOdat — APPROVED** [review]
Besides the things pointed out by @raskad. Everything looks good from my side :)

**Comment by jasonwilliams** [comment]
I had a query around the use of “engine”. I think this was already discussed so sorry for bringing it back up. But why did we go with engine over core? I’d assume the fact it’s a js engine is implied, and core would be more consistent with both ChakraCore and JavascriptCore the Microsoft and WebKit engines respectively.

For me engine is a bit like having “interpreter” in the name.

**Review by jasonwilliams — APPROVED** [review]

File: boa_profiler/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "../test_ignore.txt",
+    "../yarn.lock",
+    "../package.json",
+    "../index.html",
+    "../tests/*",
+    "../.github/*",
+]
+
+[dependencies]
+measureme = { version = "10.0.0", optional = true }
```
[the remark this answers is not in the document]
Comment (Razican) [reply boa_profiler/Cargo.toml]: I decided to add a feature, in order to not have an unused dependency when the measure feature was not selected.

**Review by raskad — APPROVED** [review]

**Pull request merged**